runtime/bcverify/staticverify.c

```diff
@@ -1758,7 +1758,7 @@ j9bcv_verifyClassStructure (J9PortLibrary * portLib, J9CfrClassFile * classfile,
 				errorType = J9NLS_CFR_ERR_BC_METHOD_INVALID_SIG__ID;
 				goto _formatError;
 			}
-			if (isInit) {
+			if (isInit && (CFR_CONSTANT_Methodref == info->tag)) {
```

Comment:
this applies to method and interface refs,

perhaps

```
if (isInit && (CFR_CONSTANT_InterfaceMethodref != info->tag)) {
```

Reply:
With this change, the test still fails in the internal issue 147145. also, see the internal issue for the latest update.

Reply:
I made a correction

Reply:
Updated. I also had to rework the fix since `info` was being modified before the second `isInit` check. `info->tag` will always be 1 during the second `isInit` check. Will rerun internal tests after the PR is approved.